aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,394 @@
+
+## Basic usage
+
+The `HttpClientFactory` can be used directly in code to access `HttpClient` instances. The service must first be registered with the ServiceProvider:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient();
+    services.AddMvc();
+}
+```
+
+Once registered, code can accept an [IHttpClientFactory](/dotnet/api/microsoft.extensions.http.ihttpclientfactory) anywhere services can be injected with [dependency injection](xref:fundamentals/dependency-injection) (DI). The `IHttpClientFactory` can be used to create a `HttpClient` instance:
+
+```csharp
+public class MyController : Controller
+{
+    IHttpClientFactory _httpClientFactory;
+
+    public MyController(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<IActionResult> Index()
+    {
+        var client = _httpClientFactory.CreateClient();
+        var result = await client.GetStringAsync("http://myurl/");
+        return View();
+    }
+}
+```
+
+Using `HttpClientFactory` in this fashion is a great way to begin refactoring an existing app. It has no impact on the way `HttpClient` is used. In places where `HttpClient` instances are currently created, replace those with a call to `CreateClient`.
+
+## Named clients
+
+If an app requires multiple distinct uses of `HttpClient`, each with different configurations, an option is to use **named clients**. Any common configuration for a named `HttpClient` can be specified during registration.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient("github", c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // GitHub API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample");   // GitHub requires a user-agent
+    });
+    services.AddHttpClient();
+}
+```
+
+In the preceding code, `AddHttpClient` is called twice: once with the name "github" and once without. The GitHub-specific client has some default configuration applied&mdash;namely the base address and two headers required to work with the GitHub API.
+
+Each time `CreateClient` is called, a new instance of `HttpClient` is created and the configuration function is called.
+
+To consume a named client, a string parameter can be passed to `CreateClient`, specifying the name of the client to be created:
+
+```csharp
+public class MyController : Controller
+{
+    IHttpClientFactory _httpClientFactory;
+
+    public MyController(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public IActionResult Index()
+    {
+        var defaultClient = _httpClientFactory.CreateClient();
+        var gitHubClient = _httpClientFactory.CreateClient("github");
+        return View();
```

Comment:
It looks like the wrong line got replaced here, 2 clients are created and neither is used.

Reply:
If you intended to create two clients here, I have a bit of a gripe with it because we'd never tell a user to do this 😆

Reply:
I originally took this sample the text below it from @glennc's work on the Wiki. It appeared the intent was to show both options being used within one app.

Happy to factor this out of the final sample app though and focus on each use case in isolation. In the sample the current approach is to register multiple clients to show the various consumption patterns. Do we need to consider that when designing the sample?

Reply:
I think registering multiple clients is fine, my specific gripe is about this method ...